### DISCUSSION TITLE
ebusd installation on raspi 3 with bullseye debian linux

### DISCUSSION THREAD

**Discussion by Kifo2023** [john30/ebusd#913] [General]
I am running a rauspi 3 (armhf) with bullseye linux.

I followed the [instation instructions](https://github.com/john30/ebusd-debian/blob/master/README.md), but got many messages about missing dependencies. I am not shure if dbus was correctly installed.
See below:

pi@raspberrypi:~$ sudo apt-get install ebusd
Reading package lists... Done
Building dependency tree
Reading state information... Done
ebusd is already the newest version.
You might want to run 'apt-get -f install' to correct these:
The following packages have unmet dependencies:
 libc6-dbg : Depends: libc6 (= 2.19-18+deb8u10) but 2.31-13+rpi1+deb11u6 is to be installed
 libc6-dev : Depends: libnsl-dev but it is not going to be installed
 libnih1 : Depends: libc6 (< 2.20) but 2.31-13+rpi1+deb11u6 is to be installed
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).

Starting the service fails:

pi@raspberrypi:~$ systemctl status ebusd.service
● ebusd.service - ebusd, the daemon for communication with eBUS heating systems.
   Loaded: loaded (/lib/systemd/system/ebusd.service; disabled)
   Active: activating (auto-restart) (Result: exit-code) since Sat 2023-05-27 18:48:47 CEST; 23s ago
  Process: 8509 ExecStart=/usr/bin/ebusd $EBUSD_OPTS (code=exited, status=1/FAILURE)

Checking with ebusd -? returns
ebusd: /usr/lib/arm-linux-gnueabihf/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by ebusd)
ebusd: /usr/lib/arm-linux-gnueabihf/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by ebusd)
ebusd: /usr/lib/arm-linux-gnueabihf/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by ebusd)

pi@raspberrypi:~$ journalctl -xn
No journal files were found.

I'm lost after spending hours with this now. Can anybody help?
Thanks in advance!
Kifo

**Comment by QubaB** [comment]
run ebusd from command line and not as a service
`ebusd -f ...`
it can probably provide more information.

**Comment by Kifo2023** [comment]
I tried so before, same result:

pi@raspberrypi:~$ ebusd -f
ebusd: /usr/lib/arm-linux-gnueabihf/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by ebusd)
ebusd: /usr/lib/arm-linux-gnueabihf/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by ebusd)
ebusd: /usr/lib/arm-linux-gnueabihf/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by ebusd)

Which other information might be helpful in this case?

**Reply by QubaB** [reply]
No idea how to solve this. I personally compiled ebusd from source.

**Comment by Kifo2023** [comment]
I am not very familiar with Linux. I tried to compile ebusd myself, but failed. I am still getting the message about missing dependencies.
I tried to follow the instruction under https://github.com/john30/ebusd/wiki/1.-Build-and-install, but was stuck already at the first step:

dpkg install git autoconf automake g++ make libssl-dev
resulted in
dpkg: error: need an action option

**Comment by Kifo2023** [comment]
I also tried to update my system. Here I get:
W: Failed to fetch https://repo.ebusd.eu/apt/default/bullseye/dists/bullseye/main/binary-armhf/Packages  server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none

W: Failed to fetch http://repo.mosquitto.org/debian/dists/jessie/main/binary-armhf/Packages  server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none

I'm lost!

**Reply by QubaB** [reply]
This is question rather for some debian discussion.  I found [https://arstech.net/server-certificate-verification-failed-cafile-etc-ssl-certs-ca-certificates-crt-crlfile-none/](https://arstech.net/server-certificate-verification-failed-cafile-etc-ssl-certs-ca-certificates-crt-crlfile-none/) . 

Can you ask some experienced linux user to help you?

**Comment by Kifo2023** [comment]
Yes, I will do so.
Updating the certificats did not solve it.
